packages/vite/src/node/server/middlewares/transform.ts

```diff
@@ -125,13 +125,19 @@ export function transformMiddleware(
         }
       }
 
-      // warn explicit /public/ paths
-      if (url.startsWith('/public/')) {
+      const publicPath = `${server.config.publicDir.slice(
+        server.config.root.length
+      )}/`
+      // warn explicit public paths
+      if (url.startsWith(publicPath)) {
         logger.warn(
           chalk.yellow(
             `files in the public directory are served at the root path.\n` +
               `Instead of ${chalk.cyan(url)}, use ${chalk.cyan(
-                url.replace(/^\/public\//, '/')
+                url.replace(
+                  new RegExp(`^${publicPath.replace(/\//g, '\\/')}`),
+                  '/'
```

Comment:
This looks a bit complex, what do you think if we directly use a slice here? We only need to remove the starting part of the `url`.

Reply:
Yes. We don't need RegExp for the case.